File: Frontend/frontend/src/app/dto/real-time/GetAdminDataResponse.ts
export type EntryDto = {
    name: string;
    numberOfElements: number;
};

export type TopDepartureDto = {
    topDepartures: EntryDto[];
};

export type TopDestinationDto = {
    topDestinations: EntryDto[];
};

export type TopHotelsDto = {
    topHotels: EntryDto[];
};

export type TopRoomTypesDto = {
    topRoomTypes: EntryDto[];
};

export type LastTravelAgencyChangesDTO = {
    recentChanges: TravelAgencyEntryDTO[]
}

export type TravelAgencyEntryDTO = {
    eventName: string;
    idChanged: number;
    change: number;
};

export type GetAdminDataResponse = {
    topHotelsDto: TopHotelsDto;
    topRoomTypesDto: TopRoomTypesDto;
    topDepartureDto: TopDepartureDto;
    topDestinationDto: TopDestinationDto;
    lastTravelAgencyChangesDto: LastTravelAgencyChangesDTO;
};

File: Services/Models/Gate/Admin/GetAdminDataResponse.cs
using Models.Admin.DTO;
using Models.Gate.TravelAgency;

namespace Models.Gate
{
    public class GetAdminDataResponse
    {
        public TopHotelsDTO TopHotelsDto { get; set; }
        public TopRoomTypesDTO TopRoomTypesDto { get; set; }
        public TopDepartureDTO TopDepartureDto { get; set; }
        public TopDestinationDTO TopDestinationDto { get; set; }
        public LastTravelAgencyChangesResponse LastTravelAgencyChangesResponse { get; set; }
    }
}